Add CreateList component for dialogs

maismin opened the pull request:
- Add material-ui/icons and react-dropzone dependencies

- Embed Open Sans font into index.html

- Add more configurations for themes

- Add image-upload-icon for dropzone

- Add dropzone component for uploading image files

- Add a CloseIconButton component for Dialog components

- Fix FormSubmitButton to receive padding, width, height as props to
control the size of the button while retaining the color themes

maismin committed:
Add CreateList component for dialogs

- Add material-ui/icons and react-dropzone dependencies

- Embed Open Sans font into index.html

- Add more configurations for themes

- Add image-upload-icon for dropzone

- Add dropzone component for uploading image files

- Add a CloseIconButton component for Dialog components

- Fix FormSubmitButton to receive padding, width, height as props to
control the size of the button while retaining the color themes

maismin committed:
Fix CreateList component

- Replace TextField with InputBase component to match the design specs

- Override default InputBase properties in themes to match the design

- Refactor custom properties from the theme object into its own object

vic2019 commented:
I think the PlusIcon (or Card) can be separate from the CreateList component. Perhaps CreateList can either take `open` from `props` or from the redux store. (And the same can be said about the sign-up/sign-in component)

Esodey commented on client/src/components/CreateList/CreateList.js:
```diff
@@ -0,0 +1,132 @@
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="form-dialog-title"
+        classes={{ paper: classes.dialogPaper }}
```
Obviously this works. I am curious how this is different than setting className={classes.dialogPaper}?

maismin replied:
I took a closer look at the documentation and played around with the **className** prop in codesandbox. The [documentation](https://material-ui.com/customization/components/#1-specific-variation-for-a-one-time-situation) states that

> Every component provides a className property which is always applied to the root element

So the following

`className={ classes.dialogPaper }`

would apply the styles to the root element, which in this case is the Dialog button. However, this isn't what we want because it is the Paper component that controls the width and height of the container in the middle. 

With that out of the way, note that there is no Paper component inside our code, so how do we access it? The [Dialog API](https://material-ui.com/api/dialog/#css) has a CSS rule for paper that applies styles to it, and the way to pass our styles to it is through the **classes** property.

`classes={{ paper: classes.dialogPaper }}`

So in summary:
- Use **className** property to target a specific component that you have access to
- Use **classes** to style target components that you don't have direct access to but have direct access to the target's parent components

Esodey commented on client/src/components/CreateList/CreateList.js:
```diff
@@ -0,0 +1,132 @@
+            width="26rem"
+            height="6.3rem"
+            handleClick={handleSubmit}
+            disabled={!name}
```
Nice addition

Esodey reviewed (approved):
Great work. Like the additions and detail

maismin commented:
> I think the PlusIcon (or Card) can be separate from the CreateList component. Perhaps CreateList can either take `open` from `props` or from the redux store. (And the same can be said about the sign-up/sign-in component)

If the Add New List Card is separated from the CreateList component then the state that controls whether or not the Dialog component opens goes up to the parent component, which in our case is the ShoppingList component. 

The question then becomes, "Is there another component that uses this 'open'  state?" 

Pushing the state up seems unnecessary because

1. I believe the 'open' state connects only the AddNewListCard and CreateList Dialog and
2. It clutters the state of the ShoppingList component because of the previous point

Here's a link from Redux's FAQ regarding [redux and state](https://redux.js.org/faq/organizing-state#do-i-have-to-put-all-my-state-into-redux-should-i-ever-use-reacts-setstate).

maismin committed:
Add functionality to FormSubmitButton

- Rename FormSubmitButton into ThemeButton

- ThemeButton can be used as both a submit and regular button while
retaining the project themes

vic2019 commented:
I just thought that making the card part of CreateList would limit how the Dialog can be opened--it cannot be opened from a button, or multiple buttons in different places. In that way, CreateList and ShoppingLists are kind of fused together into one component. 

But either way is fine since we already have the mockup and know how the app will look like.

Edit: We probably won't implement this feature, but it might be nice in theory to be able to open the CreateList dialog while the user is checking out the details of another shopping list. The user might want to grab an item from one list and put it in a newly created list

maismin committed:
Refactor CreateList component

- into smaller, identifiable pieces consisting of the Card, Dialog, and
Form

- Update theme to override styles with the project styles for Card
component, and include background color for body

maismin commented:
> Edit: We probably won't implement this feature, but it might be nice in theory to be able to open the CreateList dialog while the user is checking out the details of another shopping list. The user might want to grab an item from one list and put it in a newly created list

That's a very good point that I hadn't thought of. I went back and split the CreateList component into CreateListCard and CreateListDialog components. The CreateList component now contains the logic and passes the handlers to CreateListCard and CreateListDialog. If we ever decide to implement that feature, we'll be ready.
